### ISSUE TITLE
Updating CONTRIBUTING documentation

### ISSUE THREAD

**Issue by Enzime** [qutebrowser/qutebrowser#2851]
Some suggested changes
- [ ] Make it more clear to run all the tests with just `tox`
- [ ] Make it more clear how to run a specific test environment
- [ ] Add information about xvfb for running in the background

**Comment by The-Compiler** [comment]
[This page](https://github.com/ghacksuserjs/ghacks-user.js/wiki/Appendix-C:-Test-Sites) has some nice test sites for HTML features, mentioning some of those might also be useful.

**Comment by reed9999** [comment]
Yes please -- I'm completely new to tox (but beginner-intermediate in Python and very familiar with unit testing) and having trouble figuring out how to run tests. If I still have issues after reading up on tox and hacking around, I'll find an appropriate place to post my support issue.

**Comment by The-Compiler** [comment]
@reed9999 Note the docs already have some [examples](https://github.com/qutebrowser/qutebrowser/blob/master/CONTRIBUTING.asciidoc#running-specific-tests) on how to run tox. Also let me point out the notice at the very beginning of it :wink:

I'm happy to help if you're still having trouble though. Feel free to open an issue here, or use the [IRC channel or mailinglist](https://github.com/qutebrowser/qutebrowser#getting-help) - whatever works for you.

**Comment by JimenezJC** [comment]
Hello!

I'm new to contributing to open source and would like to contribute to this issue. After looking at contributing.asciidoc I've noticed several changes have been made since this issue has been created. I was wondering if there has been any status updates on this issue.

**Comment by toofar** [comment]
Looking it the commit histroy these seem to be the only relevent-ish changes to that file: https://github.com/qutebrowser/qutebrowser/commit/1959a7639379ff7f88c3f35cdf99f27e20ee7403 https://github.com/qutebrowser/qutebrowser/commit/5bea9c779408409e523d9e64f15e3fe9ed0a5763
So I don't think there has been much specific progress. There is a preference to the tox py36 target thing and also a mention how to get a list of tox environments though.

**Comment by ura43** [comment]
- [] Outline a workflow for testing and reviewing open PRs
